data/crac/crac-api/src/main/java/com/powsybl/openrao/data/cracapi/Crac.java

```diff
@@ -510,4 +510,14 @@ default State getState(String contingencyId, Instant instant) {
      * Remove a NetworkAction - identified by its id - from the Crac
      */
     void removeNetworkAction(String id);
+
+    /**
+     * Get all remedial action usage limitation according to the crac creation parameters
+     */
+    Map<String, RaUsageLimits> getRaUsageLimitsPerInstant();
```

Comment:
once "inside" CRAC and RAO, there is no reason to keep IDs instead of Instants
(this applies for all usages inside RAO too)
```suggestion
    Map<Instant, RaUsageLimits> getRaUsageLimitsPerInstant();
```
also, add a getRaUsageLimits(Instant instant) method

Reply:
To achieve this, i should at some point convert String (inside CracCreationParameters) to Instant (inside Crac).
So i should check the string value to infer its InstantKind.
Is this what you had in mind ?

Reply:
you don't need to deduce the instantKind, only the instant. you can do it inside every crac creator, or in the setter inside CracImpl : when the CRAC recieves the map with string keys, it maps and stores the strings to the instant objects it carries. But I do think that it would be cleaner to create an adder, like for all other CRAC objects